pandas/tests/indexing/test_loc.py

```diff
@@ -1645,6 +1645,20 @@ def test_loc_setitem_mask_td64_series_value(self):
         assert expected == result
         tm.assert_frame_equal(df, df_copy)
 
+    @pytest.mark.parametrize("func", ["loc", "iloc"])
+    @pytest.mark.parametrize("rhs_func", [list, np.array])
+    def test_loc_setitem_boolean_list(self, func, rhs_func):
+        # GH#20438 testing specifically list key, not arraylike
+        ser = Series([0, 1, 2])
+        getattr(ser, func)[[True, False, True]] = rhs_func([5, 10])
```

Comment:
can you *also* parameterize on the indexing type as well (e.g. list / array / pd.Index)

Reply:
Index is not allowed for iloc, added the remaining ones